packages/transformer/src/IModelTransformer.ts

```diff
@@ -2972,6 +2977,8 @@ export class IModelTransformer extends IModelExportHandler {
         (sourceElemFedGuid &&
           this._queryElemIdByFedGuid(this.targetDb, sourceElemFedGuid)) ||
         // FIXME<MIKE>: describe why it's safe to assume nothing has been deleted in provenanceDb
+        // FIXME<NICK>: Is it safe to assume nothing has been deleted in provenanceDb because the provenanceDb is (most likely?) the targetDb if we've made it to this line of code? And we're processing changes
+        // in the context of the sourceDb implying there are no changes in the targetDb to process therefore no deletes in provenanceDb?
```

Comment:
`provenanceDb` is not necessarily the target, that's what `queryCanAccessProvenance` tells us, and it is false in a forward sync. I have to admit though, since we no longer do a real query here against the target, maybe we should reorganize this...
I see that in an earlier PR, maybe you replaced the query with `this.sourceDb.elements.getAspects` to get the `identifierValue`? I would prefer we run a more specific query rather than get the full props of all aspects and loop through them.

Reply:
maybe after we consider reorganizing, I can give the explanation for this FIXME lol?

Reply:
I just gave it an attempt in terms of reorganizing. Let me know if I missed what you were thinking

Reply:
much closer, have one more comment though.